server/lib/validation/joi-validator.ts

```diff
@@ -58,7 +58,13 @@ export type ValidatedRequestData<T extends JoiValidationDescriptor> = {
   body: ValidatedBodyType<T>
 }
 
-/** Validates a Koa request using Joi, returning typed values that have been normalized by Joi. */
+/**
+ * Validates a Koa request using Joi, returning typed values that have been normalized by Joi.
+ *
+ * POST and PATCH requests are using a convention where if they recieve a `jsonBody` field in a
```

Comment:
Really not a fan of this, the right way to handle this is via custom Joi extensions, not by making a magic key name.

https://github.com/hapijs/joi/issues/2037#:~:text=Array%20and%20object%20string%20coercion has an example of an extension to object to do this, but it should probably be a separate type (you'll have to look into Joi stuff, I don't know how any of this works)